xbmc/input/InputCodingTableBaiduPY.cpp

```diff
@@ -121,6 +122,30 @@ std::vector<std::wstring> CInputCodingTableBaiduPY::GetResponse(int response)
   return words;
 }
 
+bool CInputCodingTableBaiduPY::Initialize()
+{
+  CSingleLock lock(m_CS);
+  if (!IsRunning())
+  {
+    Create();
+    m_initialized = true;
+  }
+
+  return true;
+}
+
+void CInputCodingTableBaiduPY::UnInitialize()
+{
+  m_Event.Set();
+  StopThread(true);

```

Comment:
I'm not entirely sure but I believe it's better to block here than not.
Let's say we don't block and users closes and reopens keyboard it could be that the thread won't be spawned and the keyboard is non-functional.

During my testing it didn't add any perceivable lag but on slower systems it might be different.